### PR TITLE
COMMERCE-1980 Refactor commerce notifications

### PR DESCRIPTION
**Pull request by alessiorendina**

### CODE REVIEW

**Comment by alessiorendina** [comment]
ci:test

**Comment by alessiorendina** [comment]
ci:test

**Review by ethanbustad — APPROVED** [review]

**Comment by alessiorendina** [comment]
ci:test

File: commerce-notification-api/src/main/java/com/liferay/commerce/notification/type/CommerceNotificationType.java [hunk 1]
```diff
@@ -27,6 +27,10 @@
 @ProviderType
 public interface CommerceNotificationType {
```
Comment (marco-leo) [inline commerce-notification-api/src/main/java/com/liferay/commerce/notification/type/CommerceNotificationType.java:28]: I would like to add the type: CommerceNotificationType<T> . so who implements don't need to cast

File: commerce-notification-service/src/main/java/com/liferay/commerce/notification/internal/upgrade/v2_1_0/CommerceNotificationQueueEntryUpgradeProcess.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+		_addColumn(
+			CommerceNotificationQueueEntryImpl.class,
+			CommerceNotificationQueueEntryImpl.TABLE_NAME, "classNameId",
+			"LONG");
+		_addColumn(
+			CommerceNotificationQueueEntryImpl.class,
+			CommerceNotificationQueueEntryImpl.TABLE_NAME, "classPK", "LONG");
+	}
+
+	private void _addColumn(
```
Comment (marco-leo) [inline commerce-notification-service/src/main/java/com/liferay/commerce/notification/internal/upgrade/v2_1_0/CommerceNotificationQueueEntryUpgradeProcess.java:40]: @ethanbustad I do see many of this method that is just a copy and past, What do you think to create a Base Class or a Util ?  Can you take care?

File: commerce-notification-service/src/main/java/com/liferay/commerce/notification/internal/upgrade/v2_1_0/CommerceNotificationQueueEntryUpgradeProcess.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+
+/**
+ * @author Alessio Antonio Rendina
+ */
+public class CommerceNotificationQueueEntryUpgradeProcess
+	extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
```
Comment (marco-leo) [inline commerce-notification-service/src/main/java/com/liferay/commerce/notification/internal/upgrade/v2_1_0/CommerceNotificationQueueEntryUpgradeProcess.java:30]: @Commando18  We are missing to fill the data of those columns

File: commerce-notification-service/src/main/java/com/liferay/commerce/notification/service/impl/CommerceNotificationQueueEntryLocalServiceImpl.java [hunk 4]
```diff
@@ -45,7 +45,7 @@
 public class CommerceNotificationQueueEntryLocalServiceImpl
 	extends CommerceNotificationQueueEntryLocalServiceBaseImpl {
 
-	@Indexable(type = IndexableType.REINDEX)
+	@Deprecated
```
Comment (marco-leo) [inline commerce-notification-service/src/main/java/com/liferay/commerce/notification/service/impl/CommerceNotificationQueueEntryLocalServiceImpl.java:52]: @Commando18 Can we call the newer methods passing the parameters?

File: commerce-notification-service/src/main/java/com/liferay/commerce/notification/service/impl/CommerceNotificationQueueEntryServiceImpl.java [hunk 5]
```diff
@@ -81,6 +101,39 @@ public int getCommerceNotificationQueueEntriesCount(long groupId)
 			getCommerceNotificationQueueEntriesCount(groupId);
 	}
 
+	@Override
+	public int getCommerceNotificationQueueEntriesCount(
```
Comment (marco-leo) [inline commerce-notification-service/src/main/java/com/liferay/commerce/notification/service/impl/CommerceNotificationQueueEntryServiceImpl.java]: @Commando18 We don't need those methods, For example in the Order Admin UI, if you can see the order and you can see that portlet than you can see this.

File: commerce-notification-service/src/main/java/com/liferay/commerce/notification/model/impl/CommerceNotificationQueueEntryImpl.java [hunk 6]
```diff
@@ -23,4 +30,23 @@
 	public CommerceNotificationQueueEntryImpl() {
 	}
 
+	@Override
+	public String getCommerceNotificationTemplateType() throws PortalException {
```
Comment (marco-leo) [inline commerce-notification-service/src/main/java/com/liferay/commerce/notification/model/impl/CommerceNotificationQueueEntryImpl.java]: @Commando18  Avoid putting methods in the ModelImpl that calls Util, let's avoid Singleton pattern usage

**Review by marco-leo — CHANGES_REQUESTED** [review]

File: commerce-notification-service/src/main/java/com/liferay/commerce/notification/internal/upgrade/v2_1_0/CommerceNotificationQueueEntryUpgradeProcess.java (hunk 2 above)
Comment (ethanbustad) [reply commerce-notification-service/src/main/java/com/liferay/commerce/notification/internal/upgrade/v2_1_0/CommerceNotificationQueueEntryUpgradeProcess.java:40]: Yeah, that's a great idea! I'll try to do that soon.

**Comment by alessiorendina** [comment]
ci:test

**Pull request merged**